packages/noir-ethereum-api/test/ethereum/recordingClient.test.ts

```diff
@@ -41,16 +50,26 @@ describe('recordingClient', () => {
     await withTempFile(async (tempFilePath) => {
       const client = createRecordingClient(createDefaultClient());
 
-      client.getBlock({ blockNumber: 14194126n });
-      client.getProof({
-        blockNumber: 14194126n,
-        storageKeys: [],
-        address: '0xb47e3cd837dDF8e4c57f05d70ab865de6e193bbb'
-      });
+      client.getBlock(GET_BLOCK_PARAMETERS);
+      client.getProof(GET_PROOF_PARAMETERS);
       await writeObject(await client.getCalls(), tempFilePath);
 
       const savedCalls = await readObject(tempFilePath);
       expect(savedCalls).toMatchObject(EXPECTED_CALLS);
     });
   });
+
+  it('record and mock', async () => {
+    await withTempFile(async (tempFilePath) => {
+      const client = createRecordingClient(createDefaultClient());
```

Comment:
Mock `createDefaultClient` `getBlock` and `getProof`

Reply:
```ts
{ getBlock() => ....
  getProof() => ...}

Reply:
createDefaultClient in recording client replaced with mockClient

Reply:
Implemented using plain js JSON mock